File: fizzbuzz.pl
for (1..100) {
    my $divBy3 = $_ % 3 == 0;
    my $divBy5 = $_ % 5 == 0;

    if ($divBy3 && $divBy5) {
        print "FizzBuzz\n";
    }
    elsif ($divBy3) {
        print "Fizz\n";
    }
    elsif ($divBy5) {
        print "Buzz\n";
    }
    else {
        print "$_\n";
    }
}

File: fizzbuzz.ml
let fizzbuzz x =
    let div_by_3 x = x mod 3 = 0 in
    let div_by_5 x = x mod 5 = 0 in
    let div_by_both x = div_by_3 x && div_by_5 x in

    if div_by_both x then "FizzBuzz"
    else if div_by_3 x then "Fizz"
    else if div_by_5 x then "Buzz"
    else string_of_int x

let main =
    let to_100 = List.init 100 ((+) 1) in
    List.iter (fun x -> print_endline (fizzbuzz x)) to_100
